bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/device/CommandHandler.java

```diff
@@ -0,0 +1,912 @@
+            try {
+                if (cmd == OnOffType.ON) {
+                    double ramptime = getRampTime(conf, 0);
+                    int ramplevel = getRampLevel(conf, 100);
+                    byte cmd2 = encode(ramptime, ramplevel);
+                    Msg m = dev.makeStandardMessage((byte) 0x0f, getOnCmd(), cmd2, getGroup(conf));
+                    dev.enqueueMessage(m, feature);
+                    logger.debug("{}: sent ramp on to switch {} time {} level {} cmd1 {}", nm(), dev.getAddress(),
+                            ramptime, ramplevel, getOnCmd());
+                } else if (cmd == OnOffType.OFF) {
+                    double ramptime = getRampTime(conf, 0);
+                    int ramplevel = getRampLevel(conf, 0 /* ignored */);
+                    byte cmd2 = encode(ramptime, ramplevel);
+                    Msg m = dev.makeStandardMessage((byte) 0x0f, getOffCmd(), cmd2, getGroup(conf));
+                    dev.enqueueMessage(m, feature);
+                    logger.debug("{}: sent ramp off to switch {} time {} cmd1 {}", nm(), dev.getAddress(), ramptime,
+                            getOffCmd());
+                }
+                // expect to get a direct ack after this!
+            } catch (IOException e) {
+                logger.warn("{}: command send i/o error: ", nm(), e);
+            } catch (FieldException e) {
+                logger.warn("{}: command send message creation error ", nm(), e);
+            }
+        }
+
+        private int getRampLevel(InsteonChannelConfiguration conf, int defaultValue) {
+            HashMap<String, @Nullable String> params = conf.getParameters();
+            return params.containsKey("ramplevel") ? Integer.parseInt(params.get("ramplevel")) : defaultValue;
+        }
+    }
+
+    @NonNullByDefault
+    public static class ManualChangeCommandHandler extends CommandHandler {
+        ManualChangeCommandHandler(DeviceFeature f) {
+            super(f);
+        }
+
+        @Override
+        public void handleCommand(InsteonChannelConfiguration conf, Command cmd, InsteonDevice dev) {
+            try {
+                if (cmd instanceof DecimalType) {
+                    int v = ((DecimalType) cmd).intValue();
+                    int cmd1 = (v != 1) ? 0x17 : 0x18; // start or stop
+                    int cmd2 = (v == 2) ? 0x01 : 0; // up or down
+                    Msg m = dev.makeStandardMessage((byte) 0x0f, (byte) cmd1, (byte) cmd2, getGroup(conf));
+                    dev.enqueueMessage(m, feature);
+                    logger.debug("{}: cmd {} sent manual change {} {} to {}", nm(), v,
+                            (cmd1 == 0x17) ? "START" : "STOP", (cmd2 == 0x01) ? "UP" : "DOWN", dev.getAddress());
+                } else {
+                    logger.warn("{}: invalid command type: {}", nm(), cmd);
+                }
+            } catch (IOException e) {
+                logger.warn("{}: command send i/o error: ", nm(), e);
```

Comment:
This is actually here because Msg.makeMessage() throws an IOException if it can't find the message. This should never happen, except during development. Can I leave as is?

Reply:
done, created custom exception to replace IOException and left as warn.